src/Enterspeed.Delivery.Sdk/Domain/Helpers/QueryStringHelper.cs

```diff
@@ -10,21 +10,13 @@ internal class QueryStringHelper
 
         public override string ToString()
         {
-            var output = string.Empty;
-
+            var builder = new StringBuilder();
             for (var i = 0; i < _keys.Count; i++)
             {
-                if (i == 0)
-                {
-                    output += $"?{_keys[i]}={_values[i]}";
-                }
-                else
-                {
-                    output += $"&{_keys[i]}={_values[i]}";
-                }
+                builder.Append(i == 0 ? $"?{_keys[i]}={_values[i]}" : $"&{_keys[i]}={_values[i]}");
```

Comment:
Ugh really don't like the original approach with two lists. How about refactoring it to something like
```csharp
 internal class QueryStringHelper
    {
        private readonly List<KeyValuePair<string, string>> _parameters = new ();

        public override string ToString()
        {
            var keyValues = _parameters.Select(
                kvp => $"{HttpUtility.UrlEncode(kvp.Key)}={HttpUtility.UrlEncode(kvp.Value)}");
            return string.Concat("?", string.Join("&", keyValues));
        }

        public void Add(string key, string value)
        {
            if (string.IsNullOrWhiteSpace(key))
            {
                throw new ArgumentNullException(nameof(key), "Key cannot be an empty string or null");
            }

            if (string.IsNullOrWhiteSpace(value))
            {
                throw new ArgumentNullException(nameof(value), "Value cannot be an empty string or null");
            }

            _parameters.Add(new KeyValuePair<string, string>(key, value));
        }
    }
```

Reply:
Should I do this in this PR?